src/dask_awkward/lib/core.py

```diff
@@ -1493,6 +1494,187 @@ def _from_iter(obj):
+) -> PartialReductionType:
+    return reducer(chunk, axis=0, keepdims=True, mask_identity=mask_identity)
+
+
+def _concat_reducer(
+    partials: list[PartialReductionType],
+) -> ak.Array:
+    return ak.concatenate(partials, axis=0)
+
+
+def _finalise_reducer(
+    partial: PartialReductionType,
+    *,
+    reducer: Callable,
+    mask_identity: bool,
+    keepdims: bool,
+) -> ak.Array:
+    return reducer(partial, axis=0, keepdims=keepdims, mask_identity=mask_identity)
+
+
+PartialPositionalReductionType = tuple[ak.Array, ak.Array]
+
+
+# Exterior reductions need starts!
+def _chunk_reducer_positional(
+    chunk: ak.Array, start: int, *, reducer: Callable, mask_identity: bool
+) -> PartialPositionalReductionType:
+    index = reducer(chunk, axis=0, keepdims=True, mask_identity=True)
+    return index + start, chunk[index]
+
+
+# All reductions need concatenation
+def _concat_reducer_positional(
+    partials: list[PartialPositionalReductionType],
+) -> PartialPositionalReductionType:
+    partial_indices, partial_values = zip(*partials)
+    return (
+        ak.concatenate(partial_indices, axis=0),
+        ak.concatenate(partial_values, axis=0),
+    )
+
+
+# Interior reductions don't need starts
+def _tree_node_reducer_positional(
+    partial: PartialPositionalReductionType, *, reducer: Callable
+) -> PartialPositionalReductionType:
+    # partial comes from concat
+    partial_index, partial_value = partial
+    final_index_index = reducer(partial_value, keepdims=True, mask_identity=True)
+    # Indices are already absolute!
+    return (
+        partial_index[final_index_index],
+        partial_value[final_index_index],
+    )
+
+
+def _finalise_reducer_positional(
+    partial: PartialPositionalReductionType,
+    *,
+    reducer: Callable,
+    mask_identity: bool,
+    keepdims: bool,
+):
+    partial_index, partial_value = partial
+    final_index_index = reducer(partial_value, keepdims=True, mask_identity=True)
+    final_index = partial_index[final_index_index]
+
+    if not mask_identity:
+        final_index = ak.fill_none(final_index, -1, axis=0)
+
+    if not keepdims:
+        final_index = final_index[0]
+    return final_index
+
+
+def axis_0_reduction(
+    *,
+    label: str,
+    array: Array,
+    is_positional: bool,
+    keepdims: bool,
+    mask_identity: bool,
+    reducer: Callable,
+    token: str | None = None,
+    dtype: Any | None = None,
+    split_every: int | bool | None = None,
+):
+    if is_positional:
+        chunked_fn = _chunk_reducer_positional
+        tree_node_fn = _tree_node_reducer_positional
+        concat_fn = _concat_reducer_positional
+        finalize_fn = _finalise_reducer_positional
+
+        # TODO HACK!
+        from dask_awkward import from_awkward
+
+        if not array.known_divisions:
+            array.eager_compute_divisions()
+        starts = ak.Array(array.divisions)[:-1]
+        tree_node_args = [from_awkward(starts, npartitions=len(starts))]
```

Comment:
@martindurant what's the best way of doing this properly — starmapping a sequence in `map_partitions`? Do we need to extend `map_partitions` to handle this, or perhaps support `dask.array` in `map_partitions`?

Reply:
> support dask.array in map_partitions

That sounds like a bad idea

Reply:
Perhaps I should circumvent `map_partitions` and do this by hand then?